fe/fe-core/src/main/java/org/apache/doris/qe/ShortCircuitQueryContext.java

```diff
@@ -66,8 +67,15 @@ public ShortCircuitQueryContext(Planner planner, Queriable analzyedQuery) throws
         this.serializedQueryOptions = ByteString.copyFrom(
                 new TSerializer().serialize(options));
         List<TExpr> exprs = new ArrayList<>();
-        for (Expr expr : planner.getFragments().get(1).getPlanRoot().getProjectList()) {
-            exprs.add(expr.treeToThrift());
+        OlapScanNode olapScanNode = (OlapScanNode) planner.getFragments().get(1).getPlanRoot();
+        if (olapScanNode.getProjectList() != null) {
+            // project on scan node
+            exprs.addAll(olapScanNode.getProjectList().stream()
```

Comment:
should think about
```
result sink([#3, #2, #1])
+-- project([#1, #2, #3])
    +-- scan
```

Reply:
ok, i will refactor some of the code after legacy planner removed